auto_round/utils.py

```diff
@@ -806,3 +826,83 @@ def dynamic_import_inference_linear(backend, bits, group_size, sym):
+    import pkg_resources
+    try:
+        version = pkg_resources.get_distribution(library_name).version
+        return version
+    except pkg_resources.DistributionNotFound:
+        return f"{library_name} is not installed"
+
+
+def get_autogptq_packing_qlinear(backend, bits=4, group_size=128, sym=False):
+    """
+    Configures and returns a QuantLinear class based on the specified backend and parameters.
+
+    Args:
+        backend (str): The backend to be used for quantization. Supported values include "qigen", "triton", "marlin",
+                       "exllama", and "cuda".
+        bits (int, optional): The number of bits for quantization. Default is 4.
+        group_size (int, optional): The group size for quantization. Default is 128.
+        sym (bool, optional): Flag indicating whether to use symmetric quantization. Default is False.
+
+    Returns:
+        class: The dynamically imported QuantLinear class configured according to the specified parameters.
+    """
+    use_triton = True
+    disable_exllamav2 = True
+    disable_exllamav1 = False
+    disable_marlin = True
+    use_qigen = False
+    if "qigen" in backend:
+        use_triton = False
+        use_qigen = True
+    elif "triton" in backend:
+        use_triton = True
+    elif "marlin" in backend and sym:
+        use_triton = False
+        disable_marlin = False
+    elif "exllama" in backend:  ##need v1 code to export
+        use_triton = True  ##same with triton
+        disable_marlin = True
+    elif "cuda" in backend:
+        use_triton = False
+        disable_marlin = True
+        disable_exllamav2 = True
+        disable_exllamav1 = True
+    if use_triton:
+        from auto_round.export.export_to_autogptq.qlinear_triton import QuantLinear
+        return QuantLinear
+    try:
+        import auto_gptq  # pylint: disable=E0401
+    except:
+        logger.error(f"please install auto_gptq via 'pip install auto-gptq' to support exporting to {backend}")
+        exit()
+
+    from auto_gptq.utils.import_utils import dynamically_import_QuantLinear  # pylint: disable=E0401
+    version = get_library_version("auto_gptq")
+    from packaging.version import Version
+    if Version(version) <= Version("0.7.1"):
+        QuantLinear = dynamically_import_QuantLinear(
+            use_triton=use_triton,
+            desc_act=False,
+            group_size=group_size,
+            bits=bits,
+            disable_exllama=disable_exllamav1,
+            disable_exllamav2=disable_exllamav2,
+            use_qigen=use_qigen,
+            disable_marlin=disable_marlin,
+        )
+    else:
+        QuantLinear = dynamically_import_QuantLinear(
+            use_triton=use_triton,
+            desc_act=False,
+            group_size=group_size,
+            bits=bits,
+            disable_exllama=disable_exllamav1,
+            disable_exllamav2=disable_exllamav2,
+            use_qigen=use_qigen,
+            use_marlin=not disable_marlin,
```

Comment:
use_tritonv2=use_tritonv2 is not needed here?

Reply:
not supported in 0.7.1

Reply:
this is 0.7.1 above version

Reply:
no need here, this is for packing, v2 and v1 are using the same packing code